### DISCUSSION TITLE
使用模板生成的qt.widgetapp项目与vs生成的QtWidgetsApplication项目的修改设置后的行为不一致

### DISCUSSION THREAD

**Discussion by freestriker** [xmake-io/xmake#3277] [Q&A] [answered]
在使用vs生成QtWidgetsApplication项目时，默认的qdebug是输出到调试输出、cout不输出，在项目/属性/链接器/系统中将子系统从窗口 (/SUBSYSTEM:WINDOWS)修改为控制台 (/SUBSYSTEM:CONSOLE)，可将qdebug与cout输出至新出现的console窗口。
而在使用xmake生成模板qt.widgetapp并用vsxmake生成vs项目后，修改上面的设置后，qdebug依旧输出到调试输出，cout不输出。尝试在xmake.lua中添加add_cxxflags("cl::/SUBSYSTEM:CONSOLE")，生成的工程依然是窗口 (/SUBSYSTEM:WINDOWS)，修改后新出现的console依然诶有任何输出。
我不知道是否xmake是否在生成上有些区别，或者说我需要怎样调整配置才可以达到一样的效果。

**Comment by waruqi** [comment, marked as answer]
> add_cxxflags("cl::/SUBSYSTEM:CONSOLE")

这个是 ldflags ，给 linker 的，不要加到 cxxflags 编译器里面去。。

而且里面的值也不对，`cl::` 是啥？

**Comment by freestriker** [comment]
因为我看在线文档里有这样的用法：add_cxxflags("cl::/GR-")、add_cxxflags("clang_cl::/GR-")，我就以为可以这样用。
在使用add_ldflags("cl::/SUBSYSTEM:CONSOLE")后，生成工程里的设置变为/SUBSYSTEM:CONSOLE，也可以正常输出到console窗口。
但为什么像开始那样单独设置就不会正常输出。

**Reply by waruqi** [reply]
直接 add_ldflags("/SUBSYSTEM:CONSOLE") 就行了，`cl::` 是指定编译器，跨编译器才需要，而且这个是 link flags 。。cl 是编译器，也不对，msvc 的链接器是 link.exe 。。

> 但为什么像开始那样单独设置就不会正常输出。

刚不是说了么，你设置到 compiler flags 怎么可能会生效呢。。这是给链接器的。不是给编译器的

**Reply by freestriker** [reply]
但是我在vs的属性设置里改成SUBSYSTEM:CONSOLE了，也有一个console出来，就是没输出

**Reply by waruqi** [reply]
vs 设置跟 xmake.lua 有啥关系。。我只说 你刚 add_cxxflags("cl::/SUBSYSTEM:CONSOLE") 设置不对 

add_cxxflags 是编译器 flags.    `/SUBSYSTEM:CONSOLE` 是链接器 flags ，这两完全没关系。。

你把链接器 flags 设置到 编译器里面，编译器怎么可能认识
